### ISSUE TITLE
MapToDotNetValue Equivalent in LiteDB

### ISSUE THREAD

**Issue by OrkanCom** [mbdavid/LiteDB#1439]
Hi folks,

_Currently i'm using LiteDB v5.0.1_

Is there any MongoDB's "[BsonTypeMapper.MapToDotNetValue](https://api.mongodb.com/csharp/current/html/M_MongoDB_Bson_BsonTypeMapper_MapToDotNetValue.htm)" equivalent function in LiteDB?

An ex: In mongoDB, I have a MongoDB's BsonDocument and it has no pre-defined strongly typed poco class ( Let's say User.cs - a user class ) in C#, but you can get the json result like;

```csharp
ObjectId actualObjectId = MongoDB.Bson.ObjectId.Parse("507f1f77bcf86cd799439011");
FilterDefinition<BsonDocument> filter = Builders<BsonDocument>.Filter.Eq("_id", actualObjectId);

BsonDocument bsonDocument = await collection.Find(filter)
                                            .FirstOrDefaultAsync(cancellationToken: cancellationToken);

if (bsonDocument != null)
 {
     object dotnetObj = BsonTypeMapper.MapToDotNetValue(bsonDocument);
     string json = Newtonsoft.Json.JsonConvert.SerializeObject(dotnetObj);
 }
```
So it gives json results such as;
```json
{"_id": "5e391acffe35950a219db194" ,"title":"asdasd", ....
```

However if you convert LiteDB's BsonDocument to json via it's serializer, like;
```json
string json = LiteDB.JsonSerializer.Serialize(bsonDocument);
```
The jsonSerializer's output will be;
```json
{"_id":{"$oid":"5e391acffe35950a219db194"},"title":"asdasd", ....
```
So it's not usefull. Think a real-world example; You need pass plain-valid json results of documents to frontend, to other nuget libraries or other web services without converting them to your strongly-typed C# classes - if json result includes [LiteDB's extended-json fields](https://github.com/mbdavid/LiteDB/blob/11c4b533fb289baba4dcceb071f91900b5573d4a/LiteDB/Document/Json/JsonWriter.cs#L44) like "$oid", "$date", "$numberLong", "numberDecimal" etc, other javascript libraries, nuget packages, web services etc will need special **json schemas** to understand and parse them.

As a summary, if you want to return just json results, you don't need to cast them to C# poco classes;
```csharp
User user = BsonMapper.Global.ToObject<User>(document);
```
There must be a way to convert LiteDB's BsonDocuments to common json (schema 4+ maybe) without converting it to a C# model or class. MongoDB driver can remove those special mappings with BsonTypeMapper.MapToDotNetValue method.

I think LiteDB has this feature but I didn't find any method or tests to do that. 
Could you help me please ?

**Comment by lbnascimento** [comment]
@OrkanCom This feature is not currently available, but is being considered for v5.1.

**Comment by nightroman** [comment]
NB It is somewhat related to https://github.com/mbdavid/LiteDB/issues/1336
